modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -227,7 +233,13 @@ public class IgniteImpl implements Ignite {
 
         txManager = new TableTxManagerImpl(clusterSvc, new HeapLockManager());
 
-        cmgMgr = new ClusterManagementGroupManager(clusterSvc, raftMgr, restComponent);
+        cmgMgr = new ClusterManagementGroupManager(
```

Comment:
Currently there's such code (by the way it's located in incorrect place) in IgniteImpl
```
    private void waitForJoinPermission() {
        // TODO https://issues.apache.org/jira/browse/IGNITE-15114
    }
```
meaning that components after CMGManager should start after receiving NodeJoinResponse[OK].
According to
```
            raftService
                    .thenCompose(CmgRaftService::joinCluster)
                    .thenRun(() -> metaStorageNodes.complete(state.metaStorageNodes()));
```
in
 `org.apache.ignite.internal.cluster.management.ClusterManagementGroupManager#handleClusterState`

it's not implemented. Could you please fix it?

Reply:
it's not implemented on purpose, it will be done in a future PR

Reply:
Ok. It's however an important part of join logic that should be implemented in the nearest future, do you agree?

Reply:
yes, I have a special ticket for that, I've updated the TODO